Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4261,64 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
+
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
 
-        if (m_systemTickActions == 0)
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 SaveAttemptSeconds = 20;
```

Comment:
May surface this constant to the top of the file, or at least out of this function so and document what it is for so it can be adjusted (possibly) in the future